db/config.cc

```diff
@@ -686,6 +686,7 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , replace_address_first_boot(this, "replace_address_first_boot", value_status::Used, "", "Like replace_address option, but if the node has been bootstrapped successfully it will be ignored. Same as -Dcassandra.replace_address_first_boot.")
     , override_decommission(this, "override_decommission", value_status::Used, false, "Set true to force a decommissioned node to join the cluster")
     , enable_repair_based_node_ops(this, "enable_repair_based_node_ops", liveness::LiveUpdate, value_status::Used, true, "Set true to use enable repair based node operations instead of streaming based")
+    , best_effort_to_removenode(this, "best_effort_to_removenode", liveness::LiveUpdate, value_status::Used, false, "Set true to perform removenode operations in best effort mode.")
```

Comment:
The name of the variable and the explanation give no hint about what the flag does and why one would or would not use it.

We should make it part of the nodetool command flags, not a node configuration flag.

Reply:
I agree. It is hard to explain and for user to understand the option because it is quite complicated. 

I mentioned here: https://github.com/scylladb/scylla/pull/7452#issuecomment-711720737

Existing nodes know it needs to sync data for the removenode operation by gossip status REMOVING_TOKEN. We can extend REMOVING_TOKEN to contain a flag if we want safe or best effort mode. This is hack solution though.

A big picture of this problem: we may want to drop the removenode operation completely and use more safe alternative replace + decommission. 

removenode is very complicated operation than replace and decommission. It is unsafe. It is hard for users, especially for new users, to choose if they should removenode or replacing. From time to time, they choose removenode operation where replacing is a better option. 

What do you think?

Reply:
If a node is dead, replacenode may not be an option. It's easy in the cloud where there is an infinite supply of nodes, but on non-cloud, if a node dies, you may not be able to find a replacement.

Reply:
Yes. It is a good point user might not find a replacement node in some extreme cases. He can delay the replace operation until a new node is available or use the unsafe removenode ops (we give user big warning. Users might change the mind if they know the operation is unsafe). For most common cases, we can always use the replace + decommission operation. We should not sacrifice the uncommon cases for common cases.